tools/slang-cpp-extractor/parser.cpp

```diff
@@ -427,6 +419,97 @@ SlangResult Parser::_parseEnum()
     return popScope();
 }
 
+SlangResult Parser::_consumeTemplate()
+{
+    // Skip the current 'template' token.
+    m_reader.advanceToken();
+
+    // Consume everything in <>
+    SLANG_RETURN_ON_FAIL(expect(TokenType::OpLess));
+
+    {
+        Index arrowCount = 1;
+        while (true)
```

Comment:
We need to be careful with these `while(true)` loops and make sure we exit even when there is an error, e.g. when reaching the end of file.